plugins/modules/net_tools/nmcli.py

```diff
@@ -314,15 +314,23 @@
        type: str
     ip_tunnel_dev:
         description:
-            - This is used with IPIP/SIT - parent device this IPIP/SIT tunnel, can use ifname.
+            - This is used with GRE/IPIP/SIT - parent device this GRE/IPIP/SIT tunnel, can use ifname.
         type: str
     ip_tunnel_remote:
        description:
-            - This is used with IPIP/SIT - IPIP/SIT destination IP address.
+            - This is used with GRE/IPIP/SIT - GRE/IPIP/SIT destination IP address.
        type: str
     ip_tunnel_local:
        description:
-            - This is used with IPIP/SIT - IPIP/SIT local IP address.
+            - This is used with GRE/IPIP/SIT - GRE/IPIP/SIT local IP address.
+       type: str
+    ip_tunnel_input_key:
+       description:
+            - This is used with GRE - The key used for tunnel input packets
```

Comment:
Just to confirm: is this the key itself (I assume it is), or a path to the key?

Reply:
Yes, it is the key itself. GRE Key is an 32 bit integer.

>       Key (4 octets)
> 
>       The Key field contains a four octet number which was inserted by
>       the encapsulator.  It may be used by the receiver to authenticate
>       the source of the packet.  The techniques for determining
>       authenticity are outside of the scope of this document.  The Key
>       field is only present if the Key Present field is set to 1.

https://netzikon.net/rfc/RFC1701-Generic-Routing-Encapsulation-GRE.html


```
+------------+------------+---------------+--------------------------------------------------------------------------------------------------------------------------------------+
|  Key Name  | Value Type | Default Value |                                                          Value Description                                                           |
+------------+------------+---------------+--------------------------------------------------------------------------------------------------------------------------------------+
| input-key  | string     |               | The key used for tunnel input packets; the property is valid only for certain tunnel modes (GRE, IP6GRE). If empty, no key is used.  |
| output-key | string     |               | The key used for tunnel output packets; the property is valid only for certain tunnel modes (GRE, IP6GRE). If empty, no key is used. |
+------------+------------+---------------+--------------------------------------------------------------------------------------------------------------------------------------+
```



https://people.freedesktop.org/~lkundrak/nm-docs/nm-settings.html